src/main/java/org/prebid/server/validation/RequestValidator.java

```diff
@@ -508,13 +564,50 @@ private void validateNativeVideoProtocols(List<Integer> protocols, int impIndex,
 
     private void validateNativeVideoProtocol(Integer protocol, int impIndex, int assetIndex, int protocolIndex)
             throws ValidationException {
-        if (protocol < 0 || protocol > 10) {
+        if (protocol < Protocol.VAST10.getValue() || protocol > Protocol.DAAST10_WRAPPER.getValue()) {
             throw new ValidationException(
                     "request.imp[%d].native.request.assets[%d].video.protocols[%d] must be in the range [1, 10]."
                             + " Got %d", impIndex, assetIndex, protocolIndex, protocol);
         }
     }
 
+    private void validateNativeEventTrackers(List<EventTracker> eventTrackers, int impIndex)
+            throws ValidationException {
+
+        if (CollectionUtils.isNotEmpty(eventTrackers)) {
+            for (int eventTrackerIndex = 0; eventTrackerIndex < eventTrackers.size(); eventTrackerIndex++) {
+                validateNativeEventTracker(eventTrackers.get(eventTrackerIndex), impIndex, eventTrackerIndex);
+            }
+        }
+    }
+
+    private void validateNativeEventTracker(EventTracker eventTracker, int impIndex, int eventIndex)
+            throws ValidationException {
+        if (eventTracker != null && (eventTracker.getEvent() < EventType.IMPRESSION.getValue()
+                || eventTracker.getEvent() > EventType.VIEWABLE_VIDEO50.getValue())) {
+            throw new ValidationException(
+                    "request.imp[%d].native.request.eventtrackers[%d].event is invalid. See section 7.6: https://iabtechlab.com/wp-content/uploads/2016/07/OpenRTB-Native-Ads-Specification-Final-1.2.pdf#page=43", impIndex, eventIndex
+            );
+        }
+
+        List<Integer> methods = eventTracker.getMethods();
+
+        if (CollectionUtils.isEmpty(methods)) {
+            throw new ValidationException(
+                    "request.imp[%d].native.request.eventtrackers[%d].method is required. See section 7.7: https://iabtechlab.com/wp-content/uploads/2016/07/OpenRTB-Native-Ads-Specification-Final-1.2.pdf#page=43", impIndex, eventIndex
+            );
+        }
+
+        for (int methodIndex = 0; methodIndex < methods.size(); methodIndex++) {
+            if (methods.get(methodIndex) < EventTrackingMethod.IMAGE.getValue()
```

Comment:
Please extract `methods.get(methodIndex)` result to a local variable and mind possible `null`